[Model] Add AutoWeightsLoader support for BERT, RoBERTa

jennifurhe opened the pull request:
## Purpose

FIX (partial) https://github.com/vllm-project/vllm/issues/15697

Also, fixes a checkpoint loading error by adding `requires_grad=False` to `position_ids`. This avoids the following exception when loading weights:

```
ValueError: Following weights were not initialized from checkpoint: {'bert.embeddings.position_ids'}
```

## Test Plan

BertEmbeddingModel:
```
python3 -m vllm.entrypoints.cli.main serve sentence-transformers/all-MiniLM-L6-v2 --served-model-name bert-embeddings --trust-remote-code
```

BertForSequenceClassification:
```
python3 -m vllm.entrypoints.cli.main serve textattack/bert-base-uncased-SST-2 --served-model-name bert-sst2 --trust-remote-code
```

RobertaForSequenceClassification
```
python3 -m vllm.entrypoints.cli.main serve cardiffnlp/twitter-roberta-base-sentiment-latest --served-model-name roberta-sentiment --trust-remote-code

python3 -m vllm.entrypoints.cli.main serve jinaai/jina-embeddings-v3 --served-model-name jina-v3 --trust-remote-code
```

RobertaEmbeddingModel:
```
python3 -m vllm.entrypoints.cli.main serve FacebookAI/roberta-base --served-model-name roberta-base --trust-remote-code

python3 -m vllm.entrypoints.cli.main serve sentence-transformers/stsb-roberta-base-v2 --served-model-name stsb-roberta --trust-remote-code
```

## Test Result

Models load successfully

jennifurhe committed:
[Model] Use AutoWeightsLoader for BERT and fix position_ids.

Signed-off-by: Jennifer He <islandhe@gmail.com>

noooop commented on vllm/model_executor/models/bert.py:
```diff
@@ -470,26 +471,28 @@ def __init__(self, *, vllm_config: VllmConfig, prefix: str = ""):
                                         self.classifier, self.bert.pooler)
 
     def load_weights(self, weights: Iterable[tuple[str, torch.Tensor]]):
```
You need to consider the Roberta model in roberta.py as well.

panyuhe replied:
Thanks for the pointer! I've refactored the RoBERTa models, and ran the following tests to verify that the models load successfully (also updated the PR description):

RobertaForSequenceClassification:
- `cardiffnlp/twitter-roberta-base-sentiment-latest`
- `jinaai/jina-embeddings-v3`

RobertaEmbeddingModel:
- `FacebookAI/roberta-base` 
- `sentence-transformers/stsb-roberta-base-v2`

noooop replied:
You should verify all the tests under /tests/models/language/pooling. 

I am doing testing locally. 

I am also testing locally. This code involves too many models.

panyuhe commented:
> Thanks for taking this on. I was thinking if we could also simplify the logic in BertModel. There are several special cases such as:
> 
> ```
>                 if name.endswith(".bias") and name not in params_dict:
>                     continue
> ```
> 
> I think they can be refactored using AutoWeightsLoader but it could be tricky to get all models right.

Thanks for the comments! 

Looking at the bias checks like:
```
pythonif name.endswith(".bias") and name not in params_dict:
    continue
```

I was thinking these become redundant when using AutoWeightsLoader since it already filters out weights that don't have corresponding parameters in the model. For other similar checks, I was able to refactor them with AutoWeightsLoader.

The main challenge for fully migrating BertModel.load_weights() to AutoWeightsLoader is the QKV fusion logic, which requires the 3-parameter signature `weight_loader(param, weight, shard_id)` that AutoWeightsLoader doesn't currently support, so I'm planning to keep the manual loading logic there.

jennifurhe committed:
Refactor base BertModel and Roberta*Model to use AutoWeightLoader.

RobertaForSequenceClassification:
- python3 -m vllm.entrypoints.cli.main serve cardiffnlp/twitter-roberta-base-sentiment-latest --served-model-name roberta-sentiment --trust-remote-code
- python3 -m vllm.entrypoints.cli.main serve jinaai/jina-embeddings-v3 --served-model-name jina-v3 --trust-remote-code

RobertaEmbeddingMode:
- python3 -m vllm.entrypoints.cli.main serve FacebookAI/roberta-base --served-model-name roberta-base --trust-remote-code
- python3 -m vllm.entrypoints.cli.main serve sentence-transformers/stsb-roberta-base-v2 --served-model-name stsb-roberta --trust-remote-code

BertEmbeddingModel:
- python3 -m vllm.entrypoints.cli.main serve sentence-transformers/all-MiniLM-L6-v2 --served-model-name bert-embeddings --trust-remote-code

Signed-off-by:  <islandhe@gmail.com>
Signed-off-by: Jen H <islandhe@gmail.com>

panyuhe commented on vllm/model_executor/models/bert.py:
```diff
@@ -470,26 +471,28 @@ def __init__(self, *, vllm_config: VllmConfig, prefix: str = ""):
+            if name.startswith("bert."):
+                bert_weights.append((name, weight))
+            else:
+                classifier_weights.append((name, weight))
```
Thanks for the suggestion, fixed!

noooop reviewed (approved):
LGTM

Thank you for simplifying this code.

noooop commented:
@DarkLight1337 


Can we pause CI testing first and wait for my local testing? 

I actually just said that I think the code is OK, but not ready.

noooop commented:
I'm testing locally and will send you the failed tests

noooop commented:
@panyuhe 
first batch

FAILED tests/models/language/pooling/test_baai.py::test_embed_models_mteb[model_info13]
FAILED tests/models/language/pooling/test_baai.py::test_embed_models_mteb[model_info14]
FAILED tests/models/language/pooling/test_baai.py::test_embed_models_correctness[model_info13]

noooop commented:
ValueError: Following weights were not initialized from checkpoint: {'model.embeddings.position_ids'}


FAILED tests/models/language/pooling/test_scoring.py::test_cross_encoder_1_to_1[BAAI/bge-reranker-v2-m3]
FAILED tests/models/language/pooling/test_scoring.py::test_cross_encoder_1_to_N[BAAI/bge-reranker-v2-m3]
FAILED tests/models/language/pooling/test_scoring.py::test_cross_encoder_N_to_N[BAAI/bge-reranker-v2-m3]
FAILED tests/models/language/pooling/test_baai.py::test_embed_models_mteb[model_info13]
FAILED tests/models/language/pooling/test_baai.py::test_embed_models_correctness[model_info13]
FAILED tests/models/language/pooling/test_jina.py::test_rerank_models_mteb[model_info0]


ERROR tests/entrypoints/openai/test_embedding_dimensions.py::test_matryoshka[model_info1-bfloat16]
ERROR tests/entrypoints/openai/test_score.py::TestModel::test_text_1_str_text_2_list[model0-True]
ERROR tests/entrypoints/openai/test_score.py::TestModel::test_text_1_str_text_2_list[model0-False]
ERROR tests/entrypoints/openai/test_score.py::TestModel::test_text_1_list_text_2_list[model0-True]
ERROR tests/entrypoints/openai/test_score.py::TestModel::test_text_1_list_text_2_list[model0-False]
ERROR tests/entrypoints/openai/test_score.py::TestModel::test_text_1_str_text_2_str[model0-True]
ERROR tests/entrypoints/openai/test_score.py::TestModel::test_text_1_str_text_2_str[model0-False]
ERROR tests/entrypoints/openai/test_score.py::TestModel::test_score_max_model_len[model0-True]
ERROR tests/entrypoints/openai/test_score.py::TestModel::test_score_max_model_len[model0-False]



FAILED tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_mteb[model_info0]
FAILED tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_mteb[model_info5]
FAILED tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_mteb[model_info6]
FAILED tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_correctness[model_info0]
FAILED tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_correctness[model_info5]
FAILED tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_correctness[model_info6]

noooop commented:
@panyuhe 
If you have any difficulty running the tests, please leave a message and we will help you. 

It took me a long time to learn how to run these tests. (╯‵□′)╯︵┻━┻

panyuhe commented:
> @panyuhe If you have any difficulty running the tests, please leave a message and we will help you.
> 
> It took me a long time to learn how to run these tests. (╯‵□′)╯︵┻━┻
> 
> vllm is really too complicated

Thank you for helping to run the tests. I'm able to reproduce the same error messages using vllm.entrypoints.cli.main on the failing models. I'm still working on setting up the test environment to run the actual pooling tests locally, and will reach out if I run into difficulties. Thank you!

noooop commented:
You may need to install mteb[bm25s]>=1.38.11, <2 to run the tests


For all dependencies, please refer to 
- https://github.com/vllm-project/vllm/blob/main/requirements/test.in
- https://github.com/vllm-project/vllm/blob/main/requirements/test.txt

But installing everything will be very slow

jennifurhe committed:
Add back checks for model named parameters.

Also, switch position_ids to be initialized as a buffer and clean up
unused code.

Signed-off-by: Jen H <islandhe@gmail.com>

Signed-off-by:  <islandhe@gmail.com>

panyuhe commented:
@noooop All of the originally failing tests are passing now and I pasted the results of running the full pooling test suite below. Thank you for your help! 

Making an amendment to my previous statement, AutoWeightsLoader does not actually check if the weight is in the model's named parameters, so I added back the check for that.

<details>
<summary>Full pooling test results (some failing due to HW)</summary>

```
# test_output_test_baai.log
✅ vllm/tests/models/language/pooling/test_baai.py::test_embed_models_mteb[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_baai.py::test_embed_models_mteb[model_info13] - PASSED
✅ vllm/tests/models/language/pooling/test_baai.py::test_embed_models_mteb[model_info14] - PASSED
✅ vllm/tests/models/language/pooling/test_baai.py::test_embed_models_correctness[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_baai.py::test_embed_models_correctness[model_info13] - PASSED
✅ vllm/tests/models/language/pooling/test_baai.py::test_embed_models_correctness[model_info14] - PASSED
✅ vllm/tests/models/language/pooling/test_baai.py::test_rerank_models_mteb[model_info0] - PASSED

# test_output_test_classification.log
✅ vllm/tests/models/language/pooling/test_classification.py::test_models[float-jason9693/Qwen2.5-1.5B-apeach] - PASSED

# test_output_test_cross_encoder.log
✅ vllm/tests/models/language/pooling/test_cross_encoder.py::test_rerank_models_mteb[model_info0] - PASSED
🟡 vllm/tests/models/language/pooling/test_cross_encoder.py::test_rerank_models_mteb[model_info1] - FAILED_OOM

# 🟡 1 CUDA OOM failure(s) detected in test_output_test_cross_encoder.log
#   🟡 vllm/tests/models/language/pooling/test_cross_encoder.py::test_rerank_models_mteb[model_info1]
#      Attempted: 146.0 MiB
#      Available: 1.38 MiB
#      Total GPU: 7.57 GiB
#      In Use: 7.37 GiB

# test_output_test_embedding_dimensions.log
# No passed/failed test results found (skipped tests excluded)

# test_output_test_embedding.log
🟡 vllm/tests/models/language/pooling/test_embedding.py::test_models[True-BAAI/bge-multilingual-gemma2] - FAILED_OOM
🟡 vllm/tests/models/language/pooling/test_embedding.py::test_models[True-intfloat/e5-mistral-7b-instruct] - FAILED_OOM
🟡 vllm/tests/models/language/pooling/test_embedding.py::test_models[True-ssmits/Qwen2-7B-Instruct-embed-base] - FAILED_OOM
🟡 vllm/tests/models/language/pooling/test_embedding.py::test_models[False-BAAI/bge-multilingual-gemma2] - FAILED_OOM
🟡 vllm/tests/models/language/pooling/test_embedding.py::test_models[False-intfloat/e5-mistral-7b-instruct] - FAILED_OOM
🟡 vllm/tests/models/language/pooling/test_embedding.py::test_models[False-BAAI/bge-base-en-v1.5] - FAILED_OOM
✅ vllm/tests/models/language/pooling/test_embedding.py::test_models[False-sentence-transformers/all-MiniLM-L12-v2] - PASSED
✅ vllm/tests/models/language/pooling/test_embedding.py::test_models[False-intfloat/multilingual-e5-small] - PASSED
✅ vllm/tests/models/language/pooling/test_embedding.py::test_models[False-Alibaba-NLP/gte-Qwen2-1.5B-instruct] - PASSED
✅ vllm/tests/models/language/pooling/test_embedding.py::test_models[False-sentence-transformers/stsb-roberta-base-v2] - PASSED

# 🟡 6 CUDA OOM failure(s) detected in test_output_test_embedding.log
#   🟡 vllm/tests/models/language/pooling/test_embedding.py::test_models[True-BAAI/bge-multilingual-gemma2]
#      Attempted: 56.0 MiB
#      Available: 49.38 MiB
#      Total GPU: 7.57 GiB
#      In Use: 7.32 GiB
#   🟡 vllm/tests/models/language/pooling/test_embedding.py::test_models[True-intfloat/e5-mistral-7b-instruct]
#      Attempted: 250.0 MiB
#      Available: 49.38 MiB
#      Total GPU: 7.57 GiB
#      In Use: 7.32 GiB
#   🟡 vllm/tests/models/language/pooling/test_embedding.py::test_models[True-ssmits/Qwen2-7B-Instruct-embed-base]
#      Attempted: 130.0 MiB
#      Available: 129.38 MiB
#      Total GPU: 7.57 GiB
#      In Use: 7.24 GiB
#   🟡 vllm/tests/models/language/pooling/test_embedding.py::test_models[False-BAAI/bge-multilingual-gemma2]
#      Attempted: 3.42 GiB
#      Available: 129.38 MiB
#      Total GPU: 7.57 GiB
#      In Use: 7.24 GiB
#   🟡 vllm/tests/models/language/pooling/test_embedding.py::test_models[False-intfloat/e5-mistral-7b-instruct]
#      Attempted: 112.0 MiB
#      Available: 29.38 MiB
#      Total GPU: 7.57 GiB
#      In Use: 7.34 GiB
#   🟡 vllm/tests/models/language/pooling/test_embedding.py::test_models[False-BAAI/bge-base-en-v1.5]
#      Attempted: 46.0 MiB
#      Available: 29.38 MiB
#      Total GPU: 7.57 GiB
#      In Use: 7.34 GiB

# test_output_test_gritlm.log
✅ vllm/tests/models/language/pooling/test_gritlm.py::test_find_array - PASSED
🟡 vllm/tests/models/language/pooling/test_gritlm.py::test_gritlm_offline_embedding - FAILED_OOM
🔴 vllm/tests/models/language/pooling/test_gritlm.py::test_gritlm_api_server_embedding - FAILED_SERVER_CRASH
🟡 vllm/tests/models/language/pooling/test_gritlm.py::test_gritlm_offline_generate - FAILED_OOM
🔴 vllm/tests/models/language/pooling/test_gritlm.py::test_gritlm_api_server_generate - FAILED_SERVER_CRASH

# 🟡 2 CUDA OOM failure(s) detected in test_output_test_gritlm.log
#   🟡 vllm/tests/models/language/pooling/test_gritlm.py::test_gritlm_offline_embedding
#      Attempted: 224.0 MiB
#      Available: 15.38 MiB
#      Total GPU: 7.57 GiB
#      In Use: 7.35 GiB
#   🟡 vllm/tests/models/language/pooling/test_gritlm.py::test_gritlm_offline_generate
#      Attempted: 224.0 MiB
#      Available: 15.38 MiB
#      Total GPU: 7.57 GiB
#      In Use: 7.35 GiB

# 🔴 2 Server crash failure(s) detected in test_output_test_gritlm.log
#   (Likely caused by previous OOM failures crashing the server process)
#   vllm/tests/models/language/pooling/test_gritlm.py::test_gritlm_api_server_embedding
#   vllm/tests/models/language/pooling/test_gritlm.py::test_gritlm_api_server_generate

# test_output_test_gte.log
✅ vllm/tests/models/language/pooling/test_gte.py::test_embed_models_mteb[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_gte.py::test_embed_models_mteb[model_info6] - PASSED
✅ vllm/tests/models/language/pooling/test_gte.py::test_embed_models_mteb[model_info7] - PASSED
✅ vllm/tests/models/language/pooling/test_gte.py::test_embed_models_mteb[model_info8] - PASSED
🟡 vllm/tests/models/language/pooling/test_gte.py::test_embed_models_mteb[model_info9] - FAILED_OOM
✅ vllm/tests/models/language/pooling/test_gte.py::test_embed_models_mteb[model_info10] - PASSED
✅ vllm/tests/models/language/pooling/test_gte.py::test_embed_models_mteb[model_info11] - PASSED
✅ vllm/tests/models/language/pooling/test_gte.py::test_embed_models_correctness[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_gte.py::test_embed_models_correctness[model_info6] - PASSED
✅ vllm/tests/models/language/pooling/test_gte.py::test_embed_models_correctness[model_info7] - PASSED
✅ vllm/tests/models/language/pooling/test_gte.py::test_embed_models_correctness[model_info8] - PASSED
🟡 vllm/tests/models/language/pooling/test_gte.py::test_embed_models_correctness[model_info9] - FAILED_OOM
✅ vllm/tests/models/language/pooling/test_gte.py::test_embed_models_correctness[model_info10] - PASSED
✅ vllm/tests/models/language/pooling/test_gte.py::test_embed_models_correctness[model_info11] - PASSED

# 🟡 2 CUDA OOM failure(s) detected in test_output_test_gte.log
#   🟡 vllm/tests/models/language/pooling/test_gte.py::test_embed_models_mteb[model_info9]
#      Attempted: 64.0 MiB
#      Available: 1.38 MiB
#      Total GPU: 7.57 GiB
#      In Use: 7.37 GiB
#   🟡 vllm/tests/models/language/pooling/test_gte.py::test_embed_models_correctness[model_info9]
#      Attempted: 54.0 MiB
#      Available: 59.38 MiB
#      Total GPU: 7.57 GiB
#      In Use: 7.31 GiB

# test_output_test_intfloat.log
✅ vllm/tests/models/language/pooling/test_intfloat.py::test_embed_models_mteb[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_intfloat.py::test_embed_models_mteb[model_info4] - PASSED
✅ vllm/tests/models/language/pooling/test_intfloat.py::test_embed_models_correctness[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_intfloat.py::test_embed_models_correctness[model_info4] - PASSED

# test_output_test_jina.log
✅ vllm/tests/models/language/pooling/test_jina.py::test_embed_models_mteb[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_jina.py::test_embed_models_correctness[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_jina.py::test_rerank_models_mteb[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_jina.py::test_matryoshka[16-half-model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_jina.py::test_matryoshka[32-half-model_info0] - PASSED

# test_output_test_mxbai_rerank.log
✅ vllm/tests/models/language/pooling/test_mxbai_rerank.py::test_rerank_models_mteb[model_info0] - PASSED

# test_output_test_nomic.log
✅ vllm/tests/models/language/pooling/test_nomic.py::test_embed_models_mteb[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_nomic.py::test_embed_models_mteb[model_info3] - PASSED
✅ vllm/tests/models/language/pooling/test_nomic.py::test_embed_models_correctness[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_nomic.py::test_embed_models_correctness[model_info3] - PASSED

# test_output_test_nomic_max_model_len.log
✅ vllm/tests/models/language/pooling/test_nomic_max_model_len.py::test_default[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_nomic_max_model_len.py::test_default[model_info1] - PASSED
✅ vllm/tests/models/language/pooling/test_nomic_max_model_len.py::test_set_max_model_len_legal[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_nomic_max_model_len.py::test_set_max_model_len_legal[model_info1] - PASSED
✅ vllm/tests/models/language/pooling/test_nomic_max_model_len.py::test_set_max_model_len_illegal[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_nomic_max_model_len.py::test_set_max_model_len_illegal[model_info1] - PASSED
✅ vllm/tests/models/language/pooling/test_nomic_max_model_len.py::test_use_rope_scaling_legal[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_nomic_max_model_len.py::test_use_rope_scaling_legal[model_info1] - PASSED
✅ vllm/tests/models/language/pooling/test_nomic_max_model_len.py::test_use_rope_scaling_illegal[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_nomic_max_model_len.py::test_use_rope_scaling_illegal[model_info1] - PASSED

# test_output_test_qwen3_reranker.log
✅ vllm/tests/models/language/pooling/test_qwen3_reranker.py::test_rerank_models_mteb[model_info0] - PASSED

# test_output_test_reward.log
❌ vllm/tests/models/language/pooling/test_reward.py::test_prm_models[True-half-Qwen/Qwen2.5-Math-PRM-7B] - FAILED (COMPUTE_CAPABILITY)
🟡 vllm/tests/models/language/pooling/test_reward.py::test_prm_models[False-half-Qwen/Qwen2.5-Math-PRM-7B] - FAILED_OOM

# 🟡 1 CUDA OOM failure(s) detected in test_output_test_reward.log
#   🟡 vllm/tests/models/language/pooling/test_reward.py::test_prm_models[False-half-Qwen/Qwen2.5-Math-PRM-7B]
#      Attempted: 260.0 MiB
#      Available: 115.38 MiB
#      Total GPU: 7.57 GiB
#      In Use: 7.26 GiB

# test_output_test_score.log
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_text_1_str_text_2_list[model0-True] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_text_1_str_text_2_list[model0-False] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_text_1_list_text_2_list[model0-True] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_text_1_list_text_2_list[model0-False] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_text_1_str_text_2_str[model0-True] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_text_1_str_text_2_str[model0-False] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_score_max_model_len[model0-True] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_score_max_model_len[model0-False] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_text_1_str_text_2_list[model1-True] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_text_1_str_text_2_list[model1-False] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_text_1_list_text_2_list[model1-True] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_text_1_list_text_2_list[model1-False] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_text_1_str_text_2_str[model1-True] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_text_1_str_text_2_str[model1-False] - PASSED
✅ vllm/tests/entrypoints/openai/test_score.py::TestModel::test_score_max_model_len[model1-True] - PASSED

# test_output_test_scoring.log
✅ vllm/tests/models/language/pooling/test_scoring.py::test_cross_encoder_1_to_1[cross-encoder/ms-marco-MiniLM-L-6-v2] - PASSED
✅ vllm/tests/models/language/pooling/test_scoring.py::test_cross_encoder_1_to_N[cross-encoder/ms-marco-MiniLM-L-6-v2] - PASSED
✅ vllm/tests/models/language/pooling/test_scoring.py::test_cross_encoder_N_to_N[cross-encoder/ms-marco-MiniLM-L-6-v2] - PASSED
✅ vllm/tests/models/language/pooling/test_scoring.py::test_cross_encoder_1_to_1[BAAI/bge-reranker-v2-m3] - PASSED
✅ vllm/tests/models/language/pooling/test_scoring.py::test_cross_encoder_1_to_N[BAAI/bge-reranker-v2-m3] - PASSED
✅ vllm/tests/models/language/pooling/test_scoring.py::test_cross_encoder_N_to_N[BAAI/bge-reranker-v2-m3] - PASSED
✅ vllm/tests/models/language/pooling/test_scoring.py::test_embedding_1_to_1[sentence-transformers/all-MiniLM-L12-v2] - PASSED
✅ vllm/tests/models/language/pooling/test_scoring.py::test_embedding_1_to_N[sentence-transformers/all-MiniLM-L12-v2] - PASSED
✅ vllm/tests/models/language/pooling/test_scoring.py::test_embedding_N_to_N[sentence-transformers/all-MiniLM-L12-v2] - PASSED

# test_output_test_snowflake_arctic_embed.log
✅ vllm/tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_mteb[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_mteb[model_info3] - PASSED
✅ vllm/tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_mteb[model_info5] - PASSED
✅ vllm/tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_mteb[model_info6] - PASSED
✅ vllm/tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_mteb[model_info7] - PASSED
✅ vllm/tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_correctness[model_info0] - PASSED
✅ vllm/tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_correctness[model_info3] - PASSED
✅ vllm/tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_correctness[model_info5] - PASSED
✅ vllm/tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_correctness[model_info6] - PASSED
✅ vllm/tests/models/language/pooling/test_snowflake_arctic_embed.py::test_embed_models_correctness[model_info7] - PASSED

# test_output_test_truncation_control.log
✅ vllm/tests/models/language/pooling/test_truncation_control.py::test_smaller_truncation_size HTTP Error 429 thrown while requesting HEAD https://huggingface.co/sentence-transformers/all-MiniLM-L12-v2/resolve/main/config.json - PASSED
✅ vllm/tests/models/language/pooling/test_truncation_control.py::test_max_truncation_size - PASSED
✅ vllm/tests/models/language/pooling/test_truncation_control.py::test_bigger_truncation_size - PASSED
```
</details>

noooop commented:
@panyuhe 

Why is buildkite/ci/pr/language-models-test-extended-pooling stuck, and why is cudagraph_capture taking so long?

Please merge the latest code, click the panyuhe:fix-bert-autoweightsloader link under the title ([Model] Add AutoWeightsLoader support for BERT, RoBERTa #20534) to sync fork -> update branch

noooop commented:
[buildkite/ci/pr/async-engine-inputs-utils-worker-test](https://buildkite.com/vllm/ci/builds/23845#0198057b-88bc-4d07-bba2-9f84bbef3878) failure is unrelated to this PR


All tests passed, my local test also passed.

@maxdebayser Is there anything else to add?

Isotr0py commented:
Merge from main branch should fix the failing async engine test.

jennifurhe committed:
Merge branch 'vllm-project:main' into fix-bert-autoweightsloader

panyuhe commented:
Done!

jennifurhe committed:
Merge branch 'vllm-project:main' into fix-bert-autoweightsloader
